src/changes/applyChanges.ts

```diff
@@ -53,6 +52,9 @@ export function applyAcceptedRejectedChanges(
     arr!.push(changeId)
   }
 
+  // will apply node-attr-change before we start to delete nodes, as this will give us right position of nodes in the tr.doc
+  changes.sort((c1) => (c1.type === 'node-attr-change' ? -1 : 0))
```

Comment:
I'm still thinking on it as it seems to me that there maybe a better solution. I don't really like the fact that we filter not by time but by type. The position in the doc should be actually changed correctly in each transaction. You should be able to map the position in the old doc to the doc produced by the most recent transaction so that would give you the correct position on which you then should be able to apply next change.

Reply:
updated to use `updatedAt` time and I will assume the editor will prevent the user from updating node attribute after deleting the node

> The position in the doc should be actually changed correctly in each transaction. You should be able to map the position in the old doc to the doc produced by the most recent transaction so that would give you the correct position on which you then should be able to apply next change.

yes that is right, but in our case after we delete the node position will map to the wrong node as we don't have the node in the document. and if we use the old doc to compare it with the new one to determine if the node is deleted so we can ignore applying node attribute change that solution will complicate things!

Reply:
Alright, that makes sense. Thanks!